transformations/mixed_language_perturbation/transformation.py

```diff
@@ -0,0 +1,93 @@
+import random
+import string
+
+from interfaces.SentenceOperation import SentenceOperation
+from tasks.TaskTypes import TaskType
+from transformers import M2M100ForConditionalGeneration, M2M100Tokenizer
+
+"""
+Mixed Language Perturbation
+This perturbation translates words in the text from English to other languages (e.g., German). It can be used to test the robustness of a model in a multilingual setting.
+"""
+
+def translate(model, tokenizer, text, src_lang, target_lang):
+    tokenizer.src_lang = src_lang
+    encoded = tokenizer(text, return_tensors="pt")
+    generated_tokens = model.generate(**encoded, forced_bos_token_id=tokenizer.get_lang_id(target_lang))
+    res = tokenizer.batch_decode(generated_tokens, skip_special_tokens=True)
+    return res
+
+def mixed_language(model, tokenizer, text, prob_mix=0.1, src_lang="en", trg_lang="fr", seed=0):
+    random.seed(seed)
+
+    words = text.split()
+    mixed_text = ""
+    for word in words:
+        if mixed_text != "":
+            mixed_text += " "
+        if random.random() < prob_mix:
+            plain_word = word.translate(str.maketrans('', '', string.punctuation)).strip()
+
+            if plain_word == "":
+                continue
+
+            if not word[0].isupper(): # lower case
+                mixed_text += translate(model, tokenizer, plain_word, src_lang, trg_lang)[0].lower()
+            else:
+                mixed_text += translate(model, tokenizer, plain_word, src_lang, trg_lang)[0]
+
+            if word[-1] in string.punctuation:
+                mixed_text += word[-1]
+        else:
+            mixed_text += word
+
+    # print(text, mixed_text)
+    return mixed_text
+
+class MixedLanguagePerturbation(SentenceOperation):
+    tasks = [
+        TaskType.TEXT_CLASSIFICATION,
+        TaskType.TEXT_TO_TEXT_GENERATION
+    ]
+    languages = ["en"]
+    tgt_languages = ["es", "de", "fr", "zh"]
+    heavy = True
+
+    def __init__(self, seed=0, prob_mix=0.3, src_lang="en", trg_lang="de"):
+        super().__init__(seed)
+        
+        self.model = M2M100ForConditionalGeneration.from_pretrained("facebook/m2m100_418M")
```

Comment:
One small thing - please cite the paper of this model too in your readme.

Reply:
We just added the citation in the README. Thanks for the review! @kaustubhdhole